tests/test_pipeline_media.py

```diff
@@ -175,258 +149,16 @@ def test_should_remove_req_res_references_before_caching_the_results(self):
         context = getattr(info.downloaded[fp].value, "__context__", None)
         self.assertIsNone(context)
 
+    def test_subclass_getting_typeerror_without_override(self):
 
-class MockedMediaPipeline(MediaPipeline):
-    def __init__(self, *args, **kwargs):
-        super().__init__(*args, **kwargs)
-        self._mockcalled = []
-
-    def download(self, request, info):
-        self._mockcalled.append("download")
-        return super().download(request, info)
-
-    def media_to_download(self, request, info, *, item=None):
-        self._mockcalled.append("media_to_download")
-        if "result" in request.meta:
-            return request.meta.get("result")
-        return super().media_to_download(request, info)
-
-    def get_media_requests(self, item, info):
-        self._mockcalled.append("get_media_requests")
-        return item.get("requests")
-
-    def media_downloaded(self, response, request, info, *, item=None):
-        self._mockcalled.append("media_downloaded")
-        return super().media_downloaded(response, request, info)
-
-    def media_failed(self, failure, request, info):
-        self._mockcalled.append("media_failed")
-        return super().media_failed(failure, request, info)
-
-    def item_completed(self, results, item, info):
-        self._mockcalled.append("item_completed")
-        item = super().item_completed(results, item, info)
-        item["results"] = results
-        return item
-
-
-class MediaPipelineTestCase(BaseMediaPipelineTestCase):
```

Comment:
These removed tests were highly dependent on override method. Let me know if you have other suggestions to handle this